blog/_posts/2022-05-30-long-term-compatibility-plans.md

```diff
@@ -0,0 +1,87 @@
+
+More importantly, during work on 3.2, we realized that maintaining this flag may be more challenging than anticipated. With time, it will be increasingly harder to be sure that our handling of it is correct. **Thus we have decided to remove the possibility of configuring the output version altogether in Scala 3.2.**
+
+## Our solution: Scala LTS & Scala Next
+
+Finally, we have reached the agreement that the best course of action would be to split Scala development into two lines, called “Scala LTS” and “Scala Next”. Both would live as separate branches in the compiler repository. Both will have separate, but possibly synchronized releases. Porting changes from one to the other should be easy and common.
+
+Scala Next will be the line that the development of the language is taking place on. It would receive frequent minor updates and all experimental language features will live here.
+
+Scala LTS on the other hand will be the stable long-term support line. One LTS will only receive patch updates which means that all releases of the same LTS line will be forward and backward compatible in terms of output. The only changes between releases will be bug fixes, non-language changes (doctool, semanticDB, reporting), and small quality-of-life enhancements (only if we are sure we are not breaking any compatibility guarantees. After more than two years we may decide to nominate one of the releases from the Next line as the new LTS. After that, we guarantee that the old LTS will be supported for at least one additional year. That means that the time of support for each LTS release is at least three years.
+
+![Scala LTS Roadmap]({{ site.baseurl }}/resources/img/scala3-lts-roadmap.svg)
+
+### The compatibility guarantees in the new model
+
+The new model doesn’t change our compatibility guarantees. We are keeping forward and backward compatibility within a single minor version and backward compatibility across different minor versions. That means that the code compiled with Scala LTS can be used as a dependency by all Scala Next versions newer than said LTS, but not the other way around.
+
+## The (future) best practices
+
+Once the LTS & Next model is implemented, we suggest the following practices for different groups of users:
+
+### Owners of commercial projects
+
+We suggest you stick to the LTS version. It will give you the best tooling support and a stable compiler. If you are migrating a bigger project from Scala 2, you should migrate it straight to the newest LTS.
+
+You may also choose to be up to date with the latest minor version. That will allow you to use new features as soon as they are added to the language. However, you will need to be ready for frequent compiler updates. Otherwise, your project may end up with an unsupported version of the language.
+
+### Library maintainers
```

Comment:
I like the strategy and these "future" best practices, but as a library maintainer, we would benefit from best practices now.  The most aggressive dependency decides for all of us, and it has been chaotic downstream.

Until the first LTS (3.3?), is the recommendation that we stick to 3.1 (where most of us are now) or jump on 3.2?  Can we say so here?

Reply:
Right, if the implication of this post is that when 3.2 lands, 3.1 will stop getting (security) patches and thus is effectively EOL, seems to me that libraries may as well move on. OTOH, if 3.1 or something was blessed an "STS" (short-term-support?) to hold us over until the real LTS lands, that would at least get security patches or something, then it could make sense to pin.

Reply:
I wonder if and how I should put that recommendation in this post.

Reply:
> I wonder if and how I should put that recommendation in this post.

You definitely should :) it should go in the summary section: "if you are a library author, here's what you should to today".